### PR TITLE
Micro-ros connection manager on an stm32. 

### PR DESCRIPTION
**Pull request by gseqBE**
Ported to zephyr from the Arduino example [here](https://github.com/micro-ROS/micro_ros_arduino/blob/humble/examples/micro-ros_reconnection_example/micro-ros_reconnection_example.ino). 

Tested with UDP transport, NOT Serial.

Example contains a publisher /counter at 1 second.
Below console on the STM32_h743zi Nucleo shows zephyr establishing a connection with the agent on a PC. The ethernet connection is broken, which displays a "Waiting for uROS agent". When the ethernet cable is connected back up, the connection is re-established. A similar test would be to start/stop the agent on the pc with similar results.

image

### CODE REVIEW

File: apps/micro-ros_reconnection/src/main.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+#define RCSOFTCHECK(fn)                                                                                                \
+  {                                                                                                                    \
+    rcl_ret_t temp_rc = fn;                                                                                            \
+    if ((temp_rc != RCL_RET_OK))                                                                                       \
+    {                                                                                                                  \
+      printk("Failed status on line %d: %d. Continuing.\n", __LINE__, (int)temp_rc);                                   \
+    }                                                                                                                  \
+  }
+
```
Comment (pablogs9) [inline apps/micro-ros_reconnection/src/main.c]: Please clean this

Comment (gseqBE) [reply apps/micro-ros_reconnection/src/main.c]: Fixed. Autoformatter got aggressive.

**Comment by pablogs9** [comment]
@mergify backport foxy, galactic, main

**Comment by pablogs9** [comment]
@mergify backport foxy galactic main

**Pull request merged**